plugins/module_utils/turbo/server.py

```diff
@@ -30,6 +31,15 @@
 please_include_me = "bar"
 
 
+@contextlib.contextmanager
+def wrap_env(env):
+    original = os.environ.copy()
+    os.environ.update(env)
+    yield
+    os.environ.clear()
+    os.environ.update(original)
```

Comment:
seems like this may cause some issues, let's see the following use case:

```
- name: task 1
  environment:
      test_var: v1

- name: task 2
  environment:
      test_var: v2

- name: task 3
```

``task 3`` will see the env variable ``test_var`` with value ``v1`` even if it is not defined anymore.
also note that for the ``lookup`` plugin the ``os.environ`` is not updated, env variables are part of ``variables`` attribute of the ``run`` method

Reply:
Yeah, good catch. I feel like completely clearing the environment for the rest of execution seems dangerous, though. I've changed it to reset the environment right before the module is run.